src/ds/d400/d400-options.cpp

```diff
@@ -466,4 +466,72 @@ namespace librealsense
     {
         snapshot = std::make_shared<const_value_option>(get_description(), 0.f);
     }
-}
+
+    void librealsense::gyro_sensitivity_option::set( float value ) 
+    {
+        auto strong = _sensor.lock();
+        if( ! strong )
+            throw invalid_value_exception( "Hid sensor is not alive for setting" );
+
+        if( strong->is_streaming() )
+            throw invalid_value_exception( "setting this option during streaming is not allowed!" );
+
+        if(!is_valid(value))
+            throw invalid_value_exception( "set(gyro_sensitivity) failed! Invalid Gyro sensitivity resolution request "
+                                           + std::to_string( value ) );
+
+        if( auto strong = _sensor.lock() )
+        {
+            strong->set_imu_sensitivity( RS2_STREAM_GYRO, value );
+        }
+    }
+
+    float librealsense::gyro_sensitivity_option::query() const
+    {
+        if( auto strong = _sensor.lock() )
+            return strong->get_imu_sensitivity( RS2_STREAM_GYRO );
+        else
+            return -1;
+        
+    }
+
+    
+    const char * librealsense::gyro_sensitivity_option::get_value_description( float val ) const
+    {
+        switch( static_cast< int >( val ) )
+        {
+            case 0: {
```

Comment:
So I understand the user input is in range of 0-4?
So can we create an Enum for it instead of this description?
And we need somewhere to place the units
It can be
`RS_OPTION_GYRO_SENSITIVITY_61_0_DEG_SEC`?
Let's speak about it on Sunday

Reply:
Done